src/PayU/PaymentsApi/PaymentsV4/Services/HTTPClient.php

```diff
@@ -100,4 +100,42 @@ public function buildRequestHeaders($merchantCode, $orderDate, $signature)
             'Content-Type: application/json;charset=utf-8'
         ];
     }
+
+    /**
+     * @param string $url
+     * @param array $requestBody
+     * @param string $secretKey
+     * @return string
+     * @throws ConnectionException
+     */
+    public function postTokenCreationRequest(
+        $url,
+        $requestBody,
+        $secretKey
+    ) {
+        $signature = $this->hashService->generateTokenSignature($requestBody, $secretKey);
+        $requestBody['signature'] = $signature;
+
+        curl_setopt_array(
+            $this->handler,
+            array(
+                CURLOPT_URL => $url,
+                CURLOPT_POST => true,
+                CURLOPT_POSTFIELDS => http_build_query($requestBody),
+                CURLOPT_HTTPHEADER => array('Content-Type: application/x-www-form-urlencoded')
+            )
+        );
+
+        $result = curl_exec($this->handler);
+        if (curl_errno($this->handler) > 0) {
+            throw new ConnectionException(
```

Comment:
In case of 4xx or 5xx, the code will not go this flow so not exception will be thrown.  Because of this the execution will continue to \PayU\PaymentsApi\PaymentsV4\Services\ResponseParser::parseTokenJsonResponse and \PayU\PaymentsApi\PaymentsV4\Services\ResponseBuilder::buildTokenResponse and return an innacurate response

Reply:
it's okey, we want to return the response data as it is. In case of 4xx or 5xx the only difference is that $token will be null

Reply:
the same behavior is also on basic authorization v3 / v4